### DISCUSSION TITLE
Slots & single root element concept

### DISCUSSION THREAD

**Discussion by mhelaiwa** [livewire/livewire#1901] [4. General]
I am using jetstream `app.blade.php` **which has `header` slot** as a layout of my livewire components.
In my livewire component called `users` I would like to put a button inside the header that shows a modal to add a new user.
However when I do so it will brake the concept of **single root element** & nothing will work. 
Any advice to solve this problem?

![image](https://user-images.githubusercontent.com/939234/97095174-d7eb6100-1664-11eb-939b-f6392703c43e.png)

**Livewire Component**

```
<div>
    {{--Header slot of app.blade.php--}}
    <x-slot name="header">
        <span>Users Management</span>
        <x-jet-button wire:click="toggle">Add User</x-jet-button>
    </x-slot>

    {{--Users Datatable--}}
    <table id="users">
        .....
    </table>
    {{--Add User Modal--}}
    <x-jet-dialog-modal wire:model="showAddModal">
        ....
    </x-jet-dialog-modal>
</div>

```

**Comment by mhelaiwa** [comment]
My initial solution for this is to extract the header slot from the layout to each component.
But this will brake jetstream's components .. I need a livewire-ish solution :smile:

**Comment by lairg99** [comment]
Have you solved your problem? I think I have a similar challenge where I want properties to be updated in the header.

```
<div>
  <x-slot name="header">
    Todos {{ $todos->count() }}        
  </x-slot>
  
  <input ...>
  <button ...>Add todo</button>
</div>
```

Up to now I have no appropriate solution.

**Comment by Altffenser** [comment]
No solution yet? :(

**Reply by lairg99** [reply]
@Altffenser Lately, I needed to display a badge in the navigation bar - some form of counting. I ended up integrating a separate component that listens for any events.

```php
class Counter extends Component
{
    protected $listeners = [
        'compensationRequestResolved' => 'render',
    ];

    public function render(): string {
        $count = CompensationRequest::pending()->count();

        return "<span class='...'>{$count}</span>";
    }
}
```

```html
<livewire:counter/>
```

It _could_ be possible to use it inside the `x-slot` any with passing any parameter if you have an array or can't rely on Eloquent.

```html
<div>
  <x-slot name="header">
    Todos <livewire:counter :$todos/>     
  </x-slot>
  
  <input ...>
  <button ...>Add todo</button>
</div>
```

There might be other ways in Livewire v3 - or even in Volt - that I haven't tried yet.
